resources/type-system/strict-inference.md

```diff
@@ -297,12 +297,14 @@ void main() {
 
 ### Function return types
 
-Declaring a recursive local function, a top-level function, a method, a
-typedef, a generic function type, or a function-typed function parameter
-without a return type is an inference failure. The return type of non-recursive
-local functions can always be inferred from downwards or upwards inference, as
-it will have a body, and the return type of the body is known (even if there
-are inference failures within).
+Declaring a recursive local function, a top-level function, a static method, a
```

Comment:
Is a recursive function always a problem?
Should it only be if the return type of the function depends on itself, and not, say:
```dart
foo(int x) => x <= 0 ? 0 : foo(x - 1).toString().length;
```
The return type of this function is no harder to deduce than a non-recursive function.

Reply:
I thought type inference only calculated an omitted return type on a local function if that function was "non-recursive", as in, never references itself.

I don't see anything in the spec regarding this concept though...